### ISSUE TITLE
If there is no change, don't exit with an error code

### ISSUE THREAD

**Issue by jbcpollak** [go-semantic-release/semantic-release#8]
Currently when the tool runs, if there is no version change detected, it exits with error code 1. This causes our CI process to record these builds as failed, but in reality they were just one of the commit codes that don't cause version changes (docs, style, etc).

I'm not sure if there is a use-case for this being considered an error.

**Comment by leifg** [comment]
The use case is: if the version didn't change, abort the deployment. Usually in CI you can do something like this:

`semantic-release || true`

This will return a zero exit code

**Comment by jbcpollak** [comment]
The problem with that is it overrides any errors that might happen with the semantic release process itself. Do different reasons exit with different values that can be tested for?

**Comment by joppa27** [comment]
TL;DR; I only run semver if certain files have changed.
I've come around on the exiting with an error code with no change to the version number. It has come in really handy when building docker images in Ci.  I use this to build docker images in Jenkins, and having an error code when there is no change is extremely helpful.

I had to be more deliberate about when I run semver though, and I use `changeset` to do that
```
        stage('get version num') {
            when {
                branch 'master'
                changeset 'images/**'
            }
            steps {
                sh 'semantic-release -vf -noci -slug arcgis/image-odjenkins -changelog CHANGELOG.md'
                script {
                    env.VERSION_NUMBER = readFile('.version').trim()
                }
            }
            post {
                success {
                    archiveArtifacts 'CHANGELOG.md'
                }
            }
}
```

**Comment by christophwitzko** [comment]
First of all sorry for the late reply. 😶

@ssorathia Thanks for trying out semantic-release. ☺️ 

I would prefer not to add this kind of a flag as the behavior could be simply achieved by adding `|| true` after the command. To distinguish between a general error or a "no new version"-error we could use different error codes (like @jbcpollak suggested).

For example:
__1__ - general error
__65__ - no new version was created
 
What are your thoughts on that?

**Comment by christophwitzko** [comment]
I think we don't break anything as it is a non-zero exit code, so no need for an additional flag.
@ssorathia would you be up for making a pull request? 😃

**Comment by ssorathia** [comment]
That makes sense.  Sure, I'll put some time into it this week.

**Comment by HazCod** [comment]
Additionally, I noticed a `.ghr` file is not being created when there is no change.
